core/sms_server_v2.py

```
@@ -154,6 +169,19 @@ async def shutdown_event():
     except Exception as e:
         logger.error(f"Error stopping workers: {e}")
     
+    # Close all open monitoring ports (v2.3)
+    if settings.monitoring_enabled:
+        try:
+            from core.admin.port_management import active_port_mappings, close_monitoring_port
+            for service in list(active_port_mappings.keys()):
+                try:
+                    close_monitoring_port(service, "system-shutdown")
+                    logger.info(f"Closed {service} port on shutdown")
```

Ruff (0.14.10)

Logging statement uses f-string

(G004)